packages/react/src/components/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,914 @@
+test('should render combobox with groups', () => {
+  render(
+    <Combobox label="label">
+      <ComboboxGroup label="Fruit">
+        <ComboboxOption>Apple</ComboboxOption>
+        <ComboboxOption>Banana</ComboboxOption>
+        <ComboboxOption>Cantaloupe</ComboboxOption>
+      </ComboboxGroup>
+      <ComboboxGroup label="Vegetables">
+        <ComboboxOption>Artichoke</ComboboxOption>
+        <ComboboxOption>Broccoli</ComboboxOption>
+        <ComboboxOption>Carrots</ComboboxOption>
+      </ComboboxGroup>
+    </Combobox>
+  );
+
+  const group1 = screen.queryAllByRole('group').at(0) as HTMLElement;
+  const group2 = screen.queryAllByRole('group').at(1) as HTMLElement;
+
+  expect(screen.queryByRole('combobox')).toBeTruthy();
+  expect(group1).toBeTruthy();
+  expect(group2).toBeTruthy();
+  expect(group1.tagName).toEqual('UL');
+  expect(group2.tagName).toEqual('UL');
+  expect(group1.getAttribute('aria-labelledby')).toBeTruthy();
+  expect(group2.getAttribute('aria-labelledby')).toBeTruthy();
+  expect(within(group1).queryAllByRole('option').at(0)?.innerText).toEqual(
+    'Apple'
+  );
+  expect(within(group1).queryAllByRole('option').at(1)?.innerText).toEqual(
+    'Banana'
+  );
+  expect(within(group1).queryAllByRole('option').at(2)?.innerText).toEqual(
+    'Cantaloupe'
+  );
+  expect(within(group2).queryAllByRole('option').at(0)?.innerText).toEqual(
+    'Artichoke'
+  );
+  expect(within(group2).queryAllByRole('option').at(1)?.innerText).toEqual(
+    'Broccoli'
+  );
+  expect(within(group2).queryAllByRole('option').at(2)?.innerText).toEqual(
+    'Carrots'
+  );
+});
+
+test('should render required combobox', () => {
+  render(
+    <Combobox label="label" required>
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(screen.getByRole('combobox').hasAttribute('required')).toEqual(true);
+  expect(screen.queryByText('Required')).toBeTruthy();
+});
+
+test('should render combobox with error', () => {
+  const errorId = 'combo-error';
+  render(
+    <Combobox
+      label="label"
+      id="combo"
+      aria-describedby="other-id"
+      required
+      error="You forgot to choose a value."
+    >
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(screen.queryByText('You forgot to choose a value.')).toBeTruthy();
+  expect(screen.getByRole('combobox').getAttribute('aria-describedby')).toBe(
+    `other-id ${errorId}`
+  );
+});
+
+test('should open combobox listbox on click', async () => {
+  render(
+    <Combobox label="label">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  assertListboxIsOpen(false);
+  fireEvent.click(screen.getByRole('combobox'));
```

Comment:
nit: I don't know what cauldron's standard practices are for this, but it's considered a best RTL practice to use the userEvents library to more closely simulate a real user. See: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event

Reply:
Yes, I'm aware.

I evaluated using userEvents here, but I think in this instance we do want `fireEvent` since we do individually test many of the different sequences of events. In this particular example, we need to test that the combobox does open on click regardless of focus. userEvents would first focus the element, then fire the click which isn't the exact sequence we want to test here.